### DISCUSSION TITLE
How do I change the size of the model viewport?

### DISCUSSION THREAD

**Discussion by mister-elliott** [google/model-viewer#1874] [Q&A]
I have taken the simplest approach to using model-viewer,  but am unsure 
how set up the html so the model will render larger
in chrome, astronaut renders in a window 300x150, using this:
`    <model-viewer src="./assets/Astronaut.glb"
                  alt="A 3D model of an astronaut"
                  background-color="#ffff00"
                  shadow-intensity="4"
                  camera-controls
                  interaction-prompt="auto"
                  auto-rotate ar magic-leap>
    </model-viewer>`

[renderdimensions.zip](https://github.com/GoogleWebComponents/model-viewer/files/3730783/renderdimensions.zip)

It would be really nice if you had a sample that was stripped down to the barest basics, and then documented how to render with minimal code at some arbitrary window size.  
Code I am using from github is a week old.
png in the zip shows the model-view seems to set the view window to 300x150.

**Comment by cdata** [comment]
@mister-elliott since `<model-viewer>` is just a DOM element, you can change its size the way you would for any other element: with CSS 🎉 

For example:

```html
<style>
model-viewer {
  width: 800px;
  height: 600px;
}
</style>
<model-viewer ...></model-viewer>
```

Will make the viewport 800x600 and scale the model appropriately.

**Reply by sampyrtuh** [reply]
Thank you so much for your help, I was looking for this and now I'm able to set the viewport size.

**Comment by mister-elliott** [comment]
nice, thanks.  when I tried to style background-color: #ffff00; in <style>model-viewer {...}, it did not take; I seem to have to set that color in the <model-viewer ....> tag.

**Comment by cdata** [comment]
Yes, unfortunately `background-color` is a special case, because rendering WebGL with transparency has consequences for things like alpha blending. We want to enable rendering in a way that would let you use `background-color` via CSS, but it will always be a trade-off.

**Comment by BR0kEN-** [comment]
Does anyone know how to make the `model-viewer` to fill the full width of the parent container and compute the height proportionally (`width: 100%; height: auto;`)?

**Reply by elalish** [reply]
It'll be the same as for any other CSS object like a `<div>`. Have you looked at calc()? https://developer.mozilla.org/en-US/docs/Web/CSS/calc()

**Reply by BR0kEN-** [reply]
I've tried setting `width: 100%; height: auto` and the model just collapsed to 0 in height.

On the https://modelviewer.dev/examples/tester.html I found the `model-viewer` has `flex-grow: 1; height: 100%;` but that works only because all ancestors of the `model-viewer` have explicitly set `height: 100%`. In the real world, I barely can find the page where all element parents define that.

https://user-images.githubusercontent.com/2760616/122530987-21435300-d030-11eb-8339-5c3196bd43b3.mov

**Reply by Karlstens** [reply]
> Does anyone know how to make the `model-viewer` to fill the full width of the parent container and compute the height proportionally (`width: 100%; height: auto;`)?

I too am experiencing issues with "height: auto". When I use "width: auto" it works just fine, displaying dynamically and as expected in different sized frames, but if I set "height: auto" then the Model-Viewer is 0px height. If I lock it at a px height, it no longer sizes dynamically. And again, if I set "height: 100%" then it fails outright with the % symbol. 

Would appreciate any help with this.

**Reply by elalish** [reply]
`height: 100%;` is what you're after; what about that is failing? You'll have to make sure the parent element (somewhere) is sized; you may need that same CSS on `<body>`.

**Reply by Karlstens** [reply]
Thanks for reaching out @elalish. Sample of what I'm loading through Wordpress below.

` <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>

<model-viewer 
bounds="tight" 
src="http://myGLB.glb" 
poster="poster.webp" 
shadow-intensity="0.8" 
shadow-softness="0.5"
auto-rotate
style="
height:auto; 
width:auto;">

</model-viewer>`

The problem is that although "width" works as expected, height will only work with a hard coded pixel value. Neither % or auto work and return a view port 0px high. 

The problem I'm trying to solve is having model-viewer act both as a thumbnail (small), and also an article header (much larger).

Example thumbnail, note how width is auto and centered, but height is hard-coded as auto height isn't working. 
![image](https://user-images.githubusercontent.com/5137574/171969930-cdef1915-b22a-45db-92ab-d6a3e2631358.png)

And a click through to the post, where again, the hardcoded height is preventing the full use of the view-port space.
![image](https://user-images.githubusercontent.com/5137574/171969990-53c00cd2-1c8d-42dc-b3be-ea9195d5da48.png)

I openly admit I'm a novice but keen to learn/understand. I'm used to working with embed solutions from either SketchFab or P3D.in, such as this example;

`<iframe allowfullscreen webkitallowfullscreen frameborder="0" seamless src="https://p3d.in/e/D0ICS+spin+load+bg-none+nopan+nozoom+norotate+controls,loader-hidden"></iframe>`

**Reply by elalish** [reply]
`height: auto;` is not going to work, as this is not an image, and so doesn't have internal dimensions. `height: 100%;` will work, but only if it is also applied to parent elements. You could also try `height: 100vh;`. Maybe try replacing MV with a `<div>` and get that to be the size you want with CSS. Then you can switch it back to MV and it'll be the same size.

**Comment by TristonH55** [comment]
Hi 
trying to set the width/height on each  model-viewer element so they can be styled to show different sizes on different  pages. 
is this possible  ?

 
    width="1800px"
    enable-pan
    id="pan-demo"
    src="/assets/test.glb"
    auto-rotate 
    camera-controls 
    min-field-of-view="1deg"

     

am not sure how i go about doing this ? i dont want to have to always use use the default <style in the Index.js.

 @elalish not sure what you meant (just like a < div >; remember to use display: block)

**Reply by elalish** [reply]
`width` is not an element attribute, it's a CSS style, same for this element and any other. So you can use `style="width: 1800px;"` or better yet set it up in your CSS style sheet: https://www.w3schools.com/cssref/pr_dim_width.asp
